subsys/dsp/Kconfig

```diff
@@ -13,9 +13,16 @@ if DSP
 config DSP_BACKEND_HAS_STATIC
 	bool
 
+config DSP_BACKEND_HAS_AGU
+	bool
+
+config DSP_BACKEND_HAS_XDATA_SECTION
+	bool
+
 choice DSP_BACKEND
 	prompt "DSP library backend selection"
 	default DSP_BACKEND_CMSIS if CMSIS_DSP
+	default DSP_BACKEND_ARCMWDT if ARC && "$(ZEPHYR_TOOLCHAIN_VARIANT)" = "arcmwdt"
```

Comment:
I would suggest moving this up one line so we choose the right optimized default (feel free to do this in another PR in order to avoid getting another round of reviews).